### ISSUE TITLE
Cleanup blob files/file data after claim update with new file

### ISSUE THREAD

**Issue by tzarebczan** [lbryio/lbry#1036]
## The Issue
When a claim is updated with a new source file, the daemon currently keeps both copies of the data in the blobfiles folder and in the file_list.  This causes bloat in the blobfiles folder since those files will never be accessed.  It also causes errors on startup in the lbrynet.file_manager.EncryptedFileManager for the invalid streamhash as well as reflector checks referencing lbry://None for the original tx. 

### Steps to reproduce
1. Publish claim
2. Edit claim with new file
3. Restart daemon

Relevant parts of the log:
Publishing:
```
2017-12-18 20:05:26,829 INFO     lbrynet.daemon.Daemon:1907: Publish: {'claim_address': None, 'name': u'lbry0191', 'bid': 0.01, 'claim_dict': {'version': '_0_0_1', 'claimType': 'streamType', 'stream': {'version': '_0_0_1', 'metadata': {u'language': u'en', u'license': u'None', u'title': u'First Pub', 'version': '_0_1_0', u'nsfw': False, u'licenseUrl': None}}}, 'file_path': u'C:\\Users\\testaccountclean\\Desktop\\LBRY_0.19.1.exe', 'change_address': None}
2017-12-18 20:05:26,851 INFO     lbrynet.daemon.Publisher:27: Starting publish for lbry0191
2017-12-18 20:05:29,157 INFO     lbrynet.lbry_file.EncryptedFileMetadataManager:246: Saved sd blob hash c1d7b0bd3a4afe2adb6c422884deeddd83d2fba4c9f5327204c61d04ebcb31b63c3f1faaf1486ca90ab1da83f762a27d to stream hash a886d6ea307b95a20faa0cf8a9e2e9221448ff3dc93bc73a6972eec1ad10ec4a1e86ddd6288dc692c8c9a3e2d2cf9ede
2017-12-18 20:05:29,282 INFO     lbrynet.core.Wallet:1384: Send claim: lbry0191 for 0.01: 080110011a7708011221080410011a0946697273742050756222002a0032044e6f6e6538004a0052005a001a50080110011a30c1d7b0bd3a4afe2adb6c422884deeddd83d2fba4c9f5327204c61d04ebcb31b63c3f1faaf1486ca90ab1da83f762a27d22186170706c69636174696f6e2f782d6d73646f776e6c6f6164 
2017-12-18 20:05:29,532 INFO     root:205: using buckets: [u'b799acbfe442fe0d9d8bd2df57718796a0ddc8e77eb46a7cc760440bf6fb86cc:1']
2017-12-18 20:05:30,000 ERROR    lbryum:116: adding signature for ff0488b21e0000000000000000001fb56d9499f61ae44cd97a9ef5d9f2dac38edf8e03318ef56aae80e86a7e57b003dbc9802954cc08d6ad6559ca85da9ccc8d8e6837f90798c95df4f79bfacb131e00000000

2017-12-18 20:05:38,859 INFO     lbrynet.core.Wallet:949: Saving metadata for claim adabd8a8a6304ea7526116f0fd097adce934153244342f2d48cbc44e3ec30fb4 0
2017-12-18 20:05:39,203 INFO     lbrynet.daemon.Daemon:717: Success! Published to lbry://lbry0191 txid: adabd8a8a6304ea7526116f0fd097adce934153244342f2d48cbc44e3ec30fb4 nout: 0
2017-12-18 20:05:39,391 INFO     lbrynet.reflector.client.client:137: Reflector needs descriptor and 26 blobs for lbry://lbry0191
2017-12-18 20:05:39,423 INFO     lbrynet.reflector.client.client:227: Sent reflector descriptor c1d7b0bd3a4afe2a
2017-12-18 20:05:56,470 INFO     lbrynet.reflector.client.client:80: Reflector has all blobs for lbry://lbry0191 (c1d7b0bd3a4afe2a)
2017-12-18 20:05:56,470 INFO     lbrynet.daemon.Daemon:713: Reflected new publication to lbry://lbry0191
```

Republishing with new file:
```
2017-12-18 20:13:37,486 INFO     lbrynet.daemon.Daemon:1907: Publish: {'claim_address': None, 'name': u'lbry0191', 'bid': 0.01, 'claim_dict': {'version': '_0_0_1', 'claimType': 'streamType', 'stream': {'version': '_0_0_1', 'metadata': {u'description': u'Uploading new file to claim', u'license': u'None', u'title': u'First Pub - update Claim', u'language': u'en', 'version': '_0_1_0', u'nsfw': False, u'licenseUrl': None}}}, 'file_path': u'C:\\Users\\testaccountclean\\Desktop\\LBRY_0.19.1.exe', 'change_address': None}
2017-12-18 20:13:37,565 INFO     lbrynet.daemon.Publisher:27: Starting publish for lbry0191
2017-12-18 20:13:40,078 INFO     lbrynet.lbry_file.EncryptedFileMetadataManager:246: Saved sd blob hash d31de14900653859dba574934c393a7b447f20607856efc1f347f54b140d9301d7de869dad0d38398f0976046c423b51 to stream hash 4dd110be18050ce5279c4ebbec0cc85af250c332dfe21efb7b1e9d0613dc9a7516a1efd7ba69dc83399ec235407e66e1
2017-12-18 20:13:40,078 INFO     lbrynet.core.Wallet:1384: Send claim: lbry0191 for 0.01: 080110011aa1010801124b080410011a18466972737420507562202d2075706461746520436c61696d221b55706c6f6164696e67206e65772066696c6520746f20636c61696d2a0032044e6f6e6538004a0052005a001a50080110011a30d31de14900653859dba574934c393a7b447f20607856efc1f347f54b140d9301d7de869dad0d38398f0976046c423b5122186170706c69636174696f6e2f782d6d73646f776e6c6f6164 
2017-12-18 20:13:40,453 INFO     root:205: using buckets: [u'adabd8a8a6304ea7526116f0fd097adce934153244342f2d48cbc44e3ec30fb4:1']
2017-12-18 20:13:42,359 ERROR    lbryum:116: adding signature for ff0488b21e0000000000000000001fb56d9499f61ae44cd97a9ef5d9f2dac38edf8e03318ef56aae80e86a7e57b003dbc9802954cc08d6ad6559ca85da9ccc8d8e6837f90798c95df4f79bfacb131e00001500

2017-12-18 20:13:43,609 ERROR    lbryum:116: adding signature for ff0488b21e0000000000000000001fb56d9499f61ae44cd97a9ef5d9f2dac38edf8e03318ef56aae80e86a7e57b003dbc9802954cc08d6ad6559ca85da9ccc8d8e6837f90798c95df4f79bfacb131e01000600

2017-12-18 20:13:52,500 INFO     lbrynet.core.Wallet:949: Saving metadata for claim 504e933da26157379eb5a3d49f315845b55aa9670563ae3054ceb3a416fe39e5 0
2017-12-18 20:13:52,703 INFO     lbrynet.daemon.Daemon:717: Success! Published to lbry://lbry0191 txid: 504e933da26157379eb5a3d49f315845b55aa9670563ae3054ceb3a416fe39e5 nout: 0
2017-12-18 20:13:52,921 INFO     lbrynet.reflector.client.client:137: Reflector needs descriptor and 26 blobs for lbry://lbry0191
2017-12-18 20:13:53,000 INFO     lbrynet.reflector.client.client:227: Sent reflector descriptor d31de14900653859
2017-12-18 20:14:08,782 INFO     lbrynet.reflector.client.client:80: Reflector has all blobs for lbry://lbry0191 (d31de14900653859)
2017-12-18 20:14:08,798 INFO     lbrynet.daemon.Daemon:713: Reflected new publication to lbry://lbry0191

```
Encrypted file manager:
`2017-12-19 06:46:36,928 ERROR    lbrynet.file_manager.EncryptedFileManager:138: An error occurred while starting a lbry file (1, a886d6ea307b95a20faa0cf8a9e2e9221448ff3dc93bc73a6972eec1ad10ec4a1e86ddd6288dc692c8c9a3e2d2cf9ede, None)`

Reflection:
```
2017-12-19 06:46:38,713 INFO     lbrynet.reflector.client.client:80: Reflector has all blobs for lbry://None (c1d7b0bd3a4afe2a)
2017-12-19 06:46:38,713 INFO     lbrynet.reflector.client.client:80: Reflector has all blobs for lbry://lbry0191 (d31de14900653859)
```

### Expected behaviour
Old blobs and file list should be cleaned up

### Actual behaviour
blobfiles still has old blobs and file exists in file_list 

## System Configuration

 

- LBRY Daemon version:
- LBRY App version:
- LBRY Installation ID:
- Operating system:

**Comment by alyssaoc** [comment]
needs a priority

**Comment by tzarebczan** [comment]
This was fixed on 0.32, confirmed that old blobs are deleted when the claim is updated.
